packages/ipfs-message-port-server/src/ipfs.ts

```diff
@@ -120,13 +127,15 @@ interface LsOptions extends AbortOptions {
   sort?: boolean
 }
 
-type LsEntry = {
+export type LsEntry = {
   name: string
+  path: string
   type: FileType
   size: number
+  depth: number
   cid: CID
-  mode: Mode
-  mtime: UnixFSTime
+  mode?: Mode
```

Comment:
mode is not optional as it has a [default value](https://github.com/ipfs/specs/blob/master/UNIXFS.md#metadata) for folders and files